### PR TITLE
Up to date check telemetry

### PR DESCRIPTION
**Pull request by panopticoncentral**
**Customer scenario**

This helps us understand how the up to date check is performing in the real world.

**Bugs this fixes:** 

#2381 

**Workarounds, if any**

None

**Risk**

Low, just adding standard telemetry

**Performance impact**

Low, just adding standard telemetry

**Is this a regression from a previous update?**

No.

### CODE REVIEW

**Comment by panopticoncentral** [comment]
@srivatsn @basoundr This was a first swipe at telemetry. I haven't done telemetry much before, so I'm not sure how much/little we should be tracking, let me know what you think.

**Comment by heejaechang** [comment]
looks fine. 

some FYI.

1. if same events happen repeatedly, probably better to cache string rather than doing s1 + s2.ToLower() every single time. if event fires rarely, then no need.

2. Event/Trace is old thing. might move to new one - https://wiki.vsdata.io/data_model_convention?s[]=usertask

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Telemetry/VsTelemetryService.cs [hunk 1]
```diff
@@ -15,6 +15,18 @@ internal class VsTelemetryService : ITelemetryService
         private const string PropertyPrefix = "VS.ProjectSystem.Managed.";
 
         /// <summary>
+        /// Post an event with the event name. This event will be correlated with the Project Telemetry Correlation asset.
```
Comment (basoundr) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Telemetry/VsTelemetryService.cs]: `This event will be correlated with the Project Telemetry Correlation asset.`. This statement is no longer true. I forgot to remove them when I made the changes, my bad.

We dont create any assets. So if possible can you remove that statement from all the doc comments? as a follow-up commit or PR?

**Comment by panopticoncentral** [comment]
@heejaechang Something like this?

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Telemetry/VsTelemetryService.cs [hunk 2]
```diff
@@ -14,9 +14,47 @@ internal class VsTelemetryService : ITelemetryService
         private const string EventPrefix = "vs/projectsystem/managed/";
         private const string PropertyPrefix = "VS.ProjectSystem.Managed.";
 
+        private readonly Dictionary<string, (string Event, Dictionary<string, string> Properties)> _eventCache = new Dictionary<string, (string, Dictionary<string, string>)>();
+
+        private (string Event, Dictionary<string, string> Properties) GetEventInfo(string eventName)
+        {
+            if (!_eventCache.TryGetValue(eventName, out var eventInfo))
```
Comment (heejaechang) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Telemetry/VsTelemetryService.cs]: is no lock okay? not used concurrently? otherwise looks good.

Comment (panopticoncentral) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Telemetry/VsTelemetryService.cs]: It could be used concurrently, will update.

**Pull request merged**